### DISCUSSION TITLE
How to monitor changes in postgresql database

### DISCUSSION THREAD

**Discussion by qingtian1771** [supabase/supabase#27452] [Questions]
I want to monitor insert, update, and delete operations in db, and when these happens, I want to update vector database according the changes, How can I to this?

**Comment by GaryAustin1** [comment]
You can use Postgres trigger functions to run code for those operations.  Not sure what "update vector database" means.  If it is in the same Postgres instance then you can do SQL operations in the functions.  Otherwise you would need to use webhooks, pg_net, or http to send data to an external endpoint.

**Reply by qingtian1771** [reply]
Thanks for the answer! 
But I don't want to disturb the DB, I don't want to insert any trigger functions. 
"Update vector database" means I will use the inserted content or updated content to update embedding and insert to vector db or update embeddings in vector db. And I don't want to make these as part of business code. it's a function support codes. 

These DB(postgresql) to DB(vector database) solution can be used by lagecy system, the integration can be done without touching the business code and database, you can only use vector database to support the business.
I think maybe kafka can do these kinds of things, but I'm learning it now.
Do you some thinking about this?

Thanks!
